core/src/main/java/org/jdbi/v3/core/statement/StatementContext.java

```diff
@@ -475,26 +478,71 @@ public void setConcurrentUpdatable(final boolean concurrentUpdatable) {
         this.concurrentUpdatable = concurrentUpdatable;
     }
 
+    /**
+     * @return the {@link Instant} at which query execution began
+     */
+    @Nullable
     public Instant getExecutionMoment() {
         return executionMoment;
     }
 
+    /**
+     * @return the {@link ZonedDateTime} at which query execution began
+     */
+    @Nullable
+    public ZonedDateTime getExecutionTimestamp() {
+        return executionMoment.atZone(getConfig(Time.class).getClock().getZone());
+    }
+
+    /**
+     * for jdbi-internal use only
+     */
     public void setExecutionMoment(Instant executionMoment) {
         this.executionMoment = executionMoment;
     }
 
+    /**
+     * @return the {@link Instant} at which query execution ended, if it did so successfully
+     */
+    @Nullable
     public Instant getCompletionMoment() {
         return completionMoment;
     }
 
+    /**
+     * @return the {@link ZonedDateTime} at which query execution ended, if it did so successfully
+     */
+    @Nullable
+    public ZonedDateTime getCompletionTimestamp() {
+        return completionMoment.atZone(getConfig(Time.class).getClock().getZone());
+    }
+
+    /**
+     * for jdbi-internal use only
+     */
     public void setCompletionMoment(Instant completionMoment) {
         this.completionMoment = completionMoment;
     }
 
+    /**
+     * @return the {@link Instant} at which query execution ended, if it did so with an exception
+     */
+    @Nullable
     public Instant getExceptionMoment() {
         return exceptionMoment;
     }
 
+    /**
+     * @return the {@link ZonedDateTime} at which query execution ended, if it did so with an exception
+     */
+    @Nullable
+    public ZonedDateTime getExceptionTimestamp() {
+        return exceptionMoment.atZone(getConfig(Time.class).getClock().getZone());
```

Comment:
This sure looks like it's going to cause a NPE if `exceptionMoment` isn't set yet.

Reply:
Good catch, sorry!